### ISSUE TITLE
App freeze

### ISSUE THREAD

**Issue by Lorenzo282** [futo-org/grayjay-android#1276]
### What happened?

Idk wether it's youtube or app itself but whenever I lose internet connection for more than 30sec circa I have to either restart the video or the whole app because it freezes and it's annoying. And it also happens when my phone switches automatically between wi-fi and mobile data, especially from wi-fi to mobile data.

### Grayjay Version

253

### What plugins are you seeing the problem on?

Youtube

### Plugin Version

194

### When do you experience the issue?

- [X] While logged in
- [ ] While logged out
- [ ] N/A

**Comment by usertoroot** [comment]
Been trying to reproduce this I did the following:
1. On WiFi start a video
2. Switch off WiFi and switch to cellular
3. Watch the video to see if any issues happen (tried for 5-10 minutes)

I was not able to reproduce. Can you please let us know which device you are using and which android version?

**Comment by Lorenzo282** [comment]
Pixel 6a androif 14
I have noticed that it only happens with mp4 file, when I choose webm I have no problems

**Comment by support-grayjay** [comment]
@Lorenzo282 apologies for the late reply. Are you still experiencing the same issue on the latest version of Grayjay?

**Comment by Lorenzo282** [comment]
Yeah, switching between qualities helps. I've continued experiencing it since after one update I've not been able to choose webm version for the most of the videos.

Sent from Proton Mail Android

-------- Original Message --------

**Comment by eightys3v3n** [comment]
More information that I think is related to this.

Download a video, close GrayJay entirely, block GrayJay's internet access somehow (see below), open GrayJay and attempt to play the downloaded video. When I perform this, the video starts playing in the background but GrayJay remains frozen indefinitely.

I have tried the following methods to block GrayJay internet access:
- use parental controls on the router to block all network traffic but keep the phone connected to the WiFi so it thinks it has a network connection.
- throttle cellular data down to tens of kbps with no WiFi.
- use an Android firewall to block GrayJay access (NetGuard).

There appears to be some network call happening that blocks the entire interface until it responds. This network call also seems to never timeout given that the app usually never becomes responsive again for me. The selected video will keep playing indefinitely if it was using an offline copy, but the interface is frozen and Android asks to kill GrayJay.

I assume this is the same thing occurring in the issue mentioned above; the internet connection switches and GrayJay continues to wait for the response from the old internet connection rather than giving up and trying to make the request again. Then if it never gets a response (connection reports as up but never works, or Android switched to a new connection and disabled the old one) GrayJay never gets over it and the interface is blocked by this waiting connection attempt.

My issue can be bypassed by shutting off cellular and WiFi (so there is no internet access reported by Android), open GrayJay and start playing a video, then re-enable cellular and WiFi. GrayJay will continue to work until for at least the rest of that video as it seems to not make that request that freezes the interface while a video is already playing.
This doesn't resolve the connection switching issue though.

**Comment by Lorenzo282** [comment]
Understandable, thank you for taking time to explain. Will keep that in mind the next time I have an issue with it.

Have a nice day and Merry Christmas 😄

-------- Original Message --------

**Comment by Lorenzo282** [comment]
Okay thanks a lot. I will look out for it.

-------- Original Message --------

**Comment by Zvonimir-FUTO** [comment]
Fixed in v312.
